### PR TITLE
Create FTA algorithm description

### PR DESCRIPTION
**Pull request by blcham**
@Matthew-Kulich @kostobog @ahmadjana 
- please check the linking to main readme 
- do both examples make sense ?

### CODE REVIEW

File: doc/fta-construction-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# FTA construction algorithm
+
+FTA tree is generated from functional component structure, i.e. considering function parts, required functions
+as well as impairing failure modes. The generated FTA includes only AND and OR gates due to the
+modeling constructs used to describe system behaviors. The algorithm implemented in this tool is described by recursive function GENERATE_FTA which represents inner nodes of FTA tree as fault events. Each fault event is described in terms of OR gate or AND
```
Comment (Matthew-Kulich) [inline doc/fta-construction-algorithm.md:5]:
```suggestion
modeling constructs used to describe system behaviors. The algorithm implemented in this tool is described by recursive function GENERATE_FTA which represents root node and some inner nodes of FTA tree as fault events. Each node that is represented by fault event is attached to OR gate or AND
```

**Review by Matthew-Kulich — CHANGES_REQUESTED** [review]
LGTM

**Comment by blcham** [comment]
@kostobog https://github.com/ahmadjana please have a look and comment even it is merged

**Comment by ahmadjana** [comment]
the readme is OK. But we shouldn't explain the examples

**Comment by blcham** [comment]
> the readme is OK. But we shouldn't explain the examples

?

**Pull request merged**